route-controller/cmd/main.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	"code.cloudfoundry.org/korifi/controllers/config"
+	routeControllers "code.cloudfoundry.org/korifi/route-controller/controllers"
+	"code.cloudfoundry.org/korifi/tools"
+	"go.uber.org/zap/zapcore"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	metricsserver "sigs.k8s.io/controller-runtime/pkg/metrics/server"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(korifiv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(gatewayv1beta1.Install(scheme))
+}
+
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	var probeAddr string
+	var cpKubeConfig string
+
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.StringVar(&cpKubeConfig, "cp-kube-config", "", "Path to the KUBECONFIG for the Korifi cluster.")
```

Comment:
Maybe I don't understand the goal correctly, but I thought the `korifi` version of the controller would not have a `cp-kube-konfig`, but it is optional. So you could disable it and bring your own closed source variant (with an additional kubeconfig).